tools/test_utils/src/SampleFileReader.cpp

```diff
@@ -1,5 +1,6 @@
 #include "SampleFileReader.hpp"
 
+#include <algorithm>
```

Comment:
Is GCC 13.1 complaining? We should bump the version on the build server.

Reply:
This is also changed in the release branch

Reply:
Do you think we should use the clang compiler on mac os? 
 I think it is the default compiler for this platform.
Then we also get the code compiled with another compiler.

Reply:
On local branch yes. Had to fix various issues that GCC overlooks. macOS build uses GCC as well.